stmhal/rtc.c

```diff
@@ -178,49 +183,61 @@ void rtc_init(void) {
     RTCHandle.Init.OutPutPolarity = RTC_OUTPUT_POLARITY_HIGH;
     RTCHandle.Init.OutPutType = RTC_OUTPUT_TYPE_OPENDRAIN;
 
-    // if LTE enabled & ready --> no need to (re-)init RTC
+    // if LSE enabled & ready --> no need to (re-)init RTC
     if ((RCC->BDCR & (RCC_BDCR_LSEON | RCC_BDCR_LSERDY)) == (RCC_BDCR_LSEON | RCC_BDCR_LSERDY)) {
         // remove Backup Domain write protection
-        #if defined(MCU_SERIES_F7)
-        PWR->CR1 |= PWR_CR1_DBP;
-        #else
-        PWR->CR |= PWR_CR_DBP;
-        #endif
+        HAL_PWR_EnableBkUpAccess();
         // Clear source Reset Flag
         __HAL_RCC_CLEAR_RESET_FLAGS();
         // provide some status information
         rtc_info |= 0x40000 | (RCC->BDCR & 7) | (RCC->CSR & 3) << 8;
         return;
+    } else if ((RCC->BDCR & RCC_BDCR_RTCSEL) == RCC_BDCR_RTCSEL_1) {
```

Comment:
What does it mean if this condition succeeds?  Can there be a comment telling the state of the MCU if we get here?